src/Backend/Modules/CommerceDelivery/Domain/Delivery/Checkout/Quote.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Backend\Modules\CommerceDelivery\Domain\Delivery\Checkout;
+
+use Backend\Modules\Commerce\Domain\ShipmentMethod\Checkout\ShipmentMethodQuote;
+use Backend\Modules\Commerce\Domain\Vat\Exception\VatNotFound;
+use Backend\Modules\Commerce\Domain\Vat\Vat;
+use Money\Money;
+
+class Quote extends ShipmentMethodQuote
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getQuote(): array
+    {
+        // General setting
+        $availablePaymentMethods = $this->shipmentMethodSettingsRepository->getSetting('CommerceDelivery', 'availablePaymentMethods');
+
+        // Country specific shipment methods
+        $shipmentMethods = $this->shipmentMethodSettingsRepository->getSetting('CommerceDelivery', 'shipmentMethods');
+        $enabledShipmentMethods = [];
+
+        foreach ($shipmentMethods as $shipmentMethod) {
+            if (!$shipmentMethod['enabled']) {
+                continue;
+            }
+
+            $enabledShipmentMethods[$shipmentMethod['label']] = [
+                'label' => "{$shipmentMethod['label']} ({$this->moneyFormatter->localizedFormatMoney($shipmentMethod['price'])})",
+                'name' => $shipmentMethod['label'],
+                'price' => $shipmentMethod['price'],
+                'vat' => $this->getVatPrice($shipmentMethod['price']),
+                'available_payment_methods' => $availablePaymentMethods,
+            ];
+        }
+
+        if ($enabledShipmentMethods) {
+            return $enabledShipmentMethods;
+        } else {
```

Comment:
Personal preference: you can remove the `else`  because the if-statement already returns early